pandas/io/tests/test_parsers.py

```diff
@@ -3753,6 +3749,150 @@ def test_single_char_leading_whitespace(self):
         tm.assert_frame_equal(result, expected)
 
 
+class TestCompression(ParserTests, tm.TestCase):
+
+    def read_csv(self, *args, **kwargs):
+        return read_csv(*args, **kwargs)
+
+    def read_table(self, *args, **kwargs):
+        return read_csv(*args, **kwargs)
+
+    def test_zip(self):
+        try:
+            import zipfile
```

Comment:
instead of making this a direct Testing class, make it ``class Compression(object)``, then add this as a mixin to the ``TestCHighMemoryParser/LowMemory``, ``TestPythonParser``, and ``TestFixedWidth``, so these routines are run for each type of parser (and don't defined ``read_csv/read_table``. That way don't have to repeat tests and they test all engines